observer_test.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestSubject_AddText(t *testing.T) {
+	inText := emptyText()
+	inText.file = NewFile("こんにちは.txt")
+
+	tests := []struct {
+		name     string
+		subject  Subject
+		expected Subject
+		in       *Text
+	}{
+		{"t0", Subject{}, Subject{
+			curtext: inText,
+			text:    map[*Text]struct{}{inText: {}},
+		}, inText},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.subject.AddText(tt.in)
+			if !reflect.DeepEqual(tt.subject, tt.expected) {
+				t.Errorf("Test: %v failed, expected: %v got: %v", tt.name, tt.expected, tt.subject)
+			}
+		})
+	}
+}
+
+func TestSubject_AllText(t *testing.T) {
+	inText := emptyText()
+	inText.file = NewFile("こんにちは.txt")
+	testData := []struct {
+		name    string
+		subject Subject
+	}{
+		{"t0", Subject{
+			curtext: inText,
+			text:    map[*Text]struct{}{inText: {}, inText: {}, inText: {}, inText: {}},
+		}},
+	}
+	t.Run("TestAllText0", func(t *testing.T) {
+		count := 0
+		testData[0].subject.AllText(func(t *Text) {
+			count++
+		})
+		if count != len(testData[0].subject.text) {
+			t.Logf("Invalid count returned, expected: %v, got : %v", len(testData[0].subject.text), count)
+
+		}
+
+	})
+}
+
+func TestSubject_DelText(t *testing.T) {
+	inText := emptyText()
+	inText.file = NewFile("こんにちは.txt")
+	testData := []struct {
+		name     string
+		subject  Subject
+		expected Subject
+		data     *Text
+	}{
+		{"t0", Subject{
+			curtext: nil,
+			text:    map[*Text]struct{}{inText: {}},
+		}, Subject{}, inText},
+	}
+	t.Run(testData[0].name, func(t *testing.T) {
+		if err := testData[0].subject.DelText(inText); err != nil {
+			t.Errorf("DelText() error = %v", err)
+		} else if reflect.DeepEqual(testData[0].subject, testData[0].expected) {
+			t.Errorf("Expected: %v, got: %v", testData[0].expected, testData[0].subject)
+		}
+	})
+}
+
+func TestSubject_HasMultipleTexts(t *testing.T) {
+	inText := emptyText()
+	inText.file = NewFile("こんにちは.txt")
+	inText2 := emptyText()
+	inText2.file = NewFile("DevilGirls.txt")
```

Comment:
I picked "Devil Girls" because it was a book from Ed Wood for the record but I see how that could be taken out of context. I'll fix it after my flight.